e2e/simple_ondatra_test.go

```diff
@@ -0,0 +1,217 @@
+package simple_ondatra_test
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"net"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/open-traffic-generator/snappi/gosnappi"
+	"github.com/openconfig/featureprofiles/topologies/binding"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/knebind/solver"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	tpb "github.com/openconfig/kne/proto/topo"
+	mpb "github.com/openconfig/magna/proto/mirror"
+)
+
+const (
+	// destinationLabel is the outer label that is used for the generated packets in MPLS flows.
+	destinationLabel = 100
+	// innerLabel is the inner label used for the generated packets in MPLS flows.
+	innerLabel = 5000
+)
+
+var (
+	// sleepTime allows a user to specify that the test should sleep after setting
+	// up all elements (configuration, gRIBI forwarding entries, traffic flows etc.).
+	sleepTime = flag.Duration("sleep", 10*time.Second, "duration for which the test should sleep after setup")
+)
+
+// intf is a simple description of an interface.
+type intf struct {
+	// Name is the name of the interface.
+	Name string
+	// MAC is the MAC address for the interface.
+	MAC string
+}
+
+var (
+	// ateSrc describes the configuration parameters for the ATE port sourcing
+	// a flow.
+	ateSrc = &intf{
+		Name: "port1",
+		MAC:  "02:00:01:01:01:01",
+	}
+
+	ateDst = &intf{
+		Name: "port2",
+		MAC:  "02:00:02:01:01:01",
+	}
+)
+
+func TestMain(m *testing.M) {
+	ondatra.RunTests(m, binding.New)
+}
+
+// configureATE interfaces configrues the source and destination ports on ATE according to the specifications
+// above. It returns the OTG configuration object.
+func configureATEInterfaces(t *testing.T, ate *ondatra.ATEDevice, srcATE, dstATE *intf) (gosnappi.Config, error) {
+	otg := ate.OTG()
+	topology := otg.NewConfig(t)
+	for _, p := range []*intf{ateSrc, ateDst} {
+		topology.Ports().Add().SetName(p.Name)
+		dev := topology.Devices().Add().SetName(p.Name)
+		eth := dev.Ethernets().Add().SetName(fmt.Sprintf("%s_ETH", p.Name))
+		eth.SetPortName(dev.Name()).SetMac(p.MAC)
+	}
+
+	c, err := topology.ToJson()
+	if err != nil {
+		return topology, err
+	}
+	t.Logf("configuration for OTG is %s", c)
+
+	otg.PushConfig(t, topology)
+	return topology, nil
+}
+
+// pushBaseConfigs pushes the base configuration to the ATE and DUT devices in
```

Comment:
where does the DUT configuration happen here?

Reply:
Apologies, there is no "DUT" in this test, I was reusing a code structure from other tests and left a stale comment.

Since this is just an e2e magna test, I replaced the "DUT" with the mirror container that is just turning traffic around. This allows us to make sure that magna works without the DUT being a potential source of errors.